run/test/specs/utils/verify_screenshots.ts

```diff
@@ -0,0 +1,62 @@
+import looksSame from 'looks-same';
+import * as fs from 'fs';
+import test from '@playwright/test';
+import * as path from 'path';
+import { DeviceWrapper } from '../../../types/DeviceWrapper';
+import { LocatorsInterface } from '../locators';
+import { StrategyExtractionObj } from '../../../types/testing';
+import { ScreenshotFileNames } from '../../../types/testing';
+import { SupportedPlatformsType } from './open_app';
+
+// Ensure only valid baseline screenshot paths are accepted
+type BaselineScreenshotPath =
+  `run/screenshots/${SupportedPlatformsType}/${ScreenshotFileNames}.png`;
+
+// The function takes a screenshot of an element and verifies it against a baseline screenshot
+// If no baseline is available, the element screenshot is retained so that it could be used as a new baseline
+// The baseline images were taken on a Pixel 6 and an iPhone 15 Pro Max
+export async function verifyElementScreenshot(
+  device: DeviceWrapper,
+  element: {
+    text?: string;
+    maxWait?: number;
+  } & (StrategyExtractionObj | LocatorsInterface),
+  baselineScreenshotPath: BaselineScreenshotPath
```

Comment:
maybe don't allow `verifyElementScreenshot` to take something else than `LocatorsInterface` until we need it to?
i.e. make `element: LocatorsInterface` instead of what it is currently